internal/core/application/wallet_service.go

```diff
@@ -169,24 +160,31 @@ func (w *walletService) InitWallet(
 			walletLastDerivedIndex := getLatestDerivationIndexForAccount(ww, domain.WalletAccount, w.explorerService)
 			marketsLastDerivedIndex := getLatestDerivationIndexForMarkets(ww, w.explorerService)
 
-			if err := initVaultAccount(v, domain.FeeAccount, feeLastDerivedIndex, w.crawlerService); err != nil {
+			feeAddresses, err := initVaultAccount(v, domain.FeeAccount, feeLastDerivedIndex)
+			if err != nil {
 				return nil, err
 			}
 			// we dont't want to let the crawler watch for WalletAccount addresses
-			if err := initVaultAccount(v, domain.WalletAccount, walletLastDerivedIndex, nil); err != nil {
+			if _, err := initVaultAccount(v, domain.WalletAccount, walletLastDerivedIndex); err != nil {
 				return nil, err
 			}
+			marketAddresses := make([]domain.AddressInfo, 0)
 			for i, m := range marketsLastDerivedIndex {
-				if err := initVaultAccount(v, domain.MarketAccountStart+i, m, w.crawlerService); err != nil {
+				addresses, err := initVaultAccount(v, domain.MarketAccountStart+i, m)
+				if err != nil {
 					return nil, err
 				}
+				marketAddresses = append(marketAddresses, addresses...)
 			}
 
+			walletAddresses = append(walletAddresses, feeAddresses...)
```

Comment:
why not doing in one single append  both feeAddresses and marketAddresses?

Reply:
the first arg of append is the destination where to add new elements... therefore it's not possible to do something like:
```
walletAddress = append(feeAddresses..., marketAddresses...)
```

https://golang.org/ref/spec#Appending_and_copying_slices